parity-clib/examples/cpp/main.cpp

```diff
@@ -14,172 +14,138 @@
+      counter -= 1;
+      break;
+    case parity_callback_type::callback_ws:
+      std::match_results<std::string_view::iterator> results;
+      std::regex is_subscription(
+          R"(\{"jsonrpc":"2.0","result":"0[xX][a-fA-F0-9]{16}","id":1\})");
+      if (std::regex_match(response.begin(), response.end(), results,
+                           is_subscription)) {
+        counter -= 1;
+      }
+      break;
+    }
+  }
 };
 
 // list of rpc queries
-const std::vector<std::string> rpc_queries {
-	"{\"method\":\"parity_versionInfo\",\"params\":[],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_getTransactionReceipt\",\"params\":[\"0x444172bef57ad978655171a8af2cfd89baa02a97fcb773067aef7794d6913fff\"],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_estimateGas\",\"params\":[{\"from\":\"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5\"}],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_getBalance\",\"params\":[\"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5\"],\"id\":1,\"jsonrpc\":\"2.0\"}"
+const std::vector<std::string> rpc_queries{
+    R"({"method":"parity_versionInfo","params":[],"id":1,"jsonrpc":"2.0"})"s,
+    R"({"method":"eth_getTransactionReceipt","params":["0x444172bef57ad978655171a8af2cfd89baa02a97fcb773067aef7794d6913fff"],"id":1,"jsonrpc":"2.0"})"s,
+    R"({"method":"eth_estimateGas","params":[{"from":"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5"}],"id":1,"jsonrpc":"2.0"})"s,
+    R"({"method":"eth_getBalance","params":["0x0066Dc48bb833d2B59f730F33952B3c29fE926F5"],"id":1,"jsonrpc":"2.0"})"s,
 };
 
 // list of subscriptions
-const std::vector<std::string> ws_subscriptions {
-	"{\"method\":\"parity_subscribe\",\"params\":[\"eth_getBalance\",[\"0xcd2a3d9f938e13cd947ec05abc7fe734df8dd826\",\"latest\"]],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"parity_subscribe\",\"params\":[\"parity_netPeers\"],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_subscribe\",\"params\":[\"newHeads\"],\"id\":1,\"jsonrpc\":\"2.0\"}"
+const std::vector<std::string> ws_subscriptions{
+    R"({"method":"parity_subscribe","params":["eth_getBalance",["0xcd2a3d9f938e13cd947ec05abc7fe734df8dd826","latest"]],"id":1,"jsonrpc":"2.0"})"s,
+    R"({"method":"parity_subscribe","params":["parity_netPeers"],"id":1,"jsonrpc":"2.0"})"s,
+    R"({"method":"eth_subscribe","params":["newHeads"],"id":1,"jsonrpc":"2.0"})"s,
 };
-
-// callback that gets invoked upon an event
-void callback(void* user_data, const char* response, size_t _len) {
-	Callback* cb = static_cast<Callback*>(user_data);
-	if (cb->type == CALLBACK_RPC) {
-		cb->counter -= 1;
-	} else if (cb->type == CALLBACK_WS) {
-		std::regex is_subscription ("\\{\"jsonrpc\":\"2.0\",\"result\":\"0[xX][a-fA-F0-9]{16}\",\"id\":1\\}");
-		if (std::regex_match(response, is_subscription) == true) {
-			cb->counter -= 1;
-		}
-	}
-}
+const std::vector<std::string> cli_args{"--no-ipc"s, "--jsonrpc-apis=all"s,
+                                        "--chain"s, "kovan"s};
```

Comment:
doesn't have to be addressed in this PR: but I suppose we should run `goerli` instead `kovan`